fuse_examples/imaging/segmentation/picai/runner.py

```diff
@@ -0,0 +1,359 @@
+    lgr = logging.getLogger("Fuse")
+    lgr.info("Fuse Inference", {"attrs": ["bold", "underline"]})
+    infer_file = os.path.join(paths["inference_dir"], infer["infer_filename"])
+    checkpoint_file = os.path.join(paths["model_dir"], infer["checkpoint"])
+    lgr.info(f"infer_filename={checkpoint_file}", {"color": "magenta"})
+
+    lgr.info("Model:", {"attrs": "bold"})
+
+    model = create_model(train["unet_kwargs"])
+    lgr.info("Model: Done", {"attrs": "bold"})
+    ## Data
+    folds = load_pickle(
+        os.path.join(paths["data_misc_dir"], paths["data_split_filename"])
+    )  # assume exists and created in train func
+
+    infer_sample_ids = []
+    for fold in infer["infer_folds"]:
+        infer_sample_ids += folds[fold]
+
+    test_dataset = PICAI.dataset(
+        paths=paths,
+        cfg=train,
+        sample_ids=infer_sample_ids,
+        reset_cache=False,
+    )
+
+    ## Create dataloader
+    infer_dataloader = DataLoader(
+        dataset=test_dataset,
+        shuffle=False,
+        drop_last=False,
+        batch_sampler=None,
+        batch_size=train["batch_size"],
+        collate_fn=CollateDefault(),
+        num_workers=infer["num_workers"],
+    )
+    # load python lightning module
+    pl_module = LightningModuleDefault.load_from_checkpoint(
+        checkpoint_file, model_dir=paths["model_dir"], model=model, map_location="cpu", strict=True
+    )
+    # set the prediction keys to extract (the ones used be the evaluation function).
+    pl_module.set_predictions_keys(["model.seg"])  # which keys to extract and dump into file
+    lgr.info("Test Data: Done", {"attrs": "bold"})
+    # create lightining trainer.
+    pl_trainer = Trainer(
+        default_root_dir=paths["model_dir"],
+        max_epochs=train["trainer"]["num_epochs"],
+        accelerator=train["trainer"]["accelerator"],
+        devices=train["trainer"]["devices"],
+        num_sanity_val_steps=-1,
+        auto_select_gpus=True,
+    )
+    # create a trainer instance
+    predictions = pl_trainer.predict(pl_module, infer_dataloader, return_predictions=True)
+
+    with open(Path(infer_file).parents[0] / "infer.pickle", "wb") as f:
+        pickle.dump(predictions, f)
+
+    # convert list of batch outputs into a dataframe
+    infer_df = convert_predictions_to_dataframe(predictions)
+    save_dataframe(infer_df, infer_file)
+
+
+######################################
+# Analyze Template
+######################################
+def run_eval(paths: NDict, infer: NDict) -> None:
+    fuse_logger_start(output_path=None, console_verbose_level=logging.INFO)
+    lgr = logging.getLogger("Fuse")
+    lgr.info("Fuse Eval", {"attrs": ["bold", "underline"]})
+
+    # create evaluator
+    evaluator = EvaluatorDefault()
+
+    # run
+    results = evaluator.eval(
+        ids=None,
+        data=os.path.join(paths["inference_dir"], infer["infer_filename"]),
+        metrics={},
```

Comment:
for now the loss is dice loss which is 1-dice.
the only metric I know for current semantic segmentation is dice.
maybe I will call the dice metric, will also ask Moshiko if it can return just the loss somehow